src/components/AnnotationViewer.vue

```diff
@@ -503,6 +534,161 @@ export default class AnnotationViewer extends Vue {
+
+      const selectionPosition = selectionAnnotationCoordinates[0];
+      const { radius, strokeWidth } = this.getAnnotationStyle(annotation);
+
+      if (annotation.shape === AnnotationShape.Point) {
+        const annotationRadius = (radius + strokeWidth) * unitsPerPixel;
+        const annotationPosition = annotationCoordinates[0];
+        return (
+          pointDistance(selectionPosition, annotationPosition) <
+          annotationRadius
+        );
+      } else if (annotation.shape === AnnotationShape.Line) {
+        // Check if click on points of the line, or on the line directly
+        const width = strokeWidth * unitsPerPixel;
+        return annotationCoordinates.reduce(
+          (isIn: boolean, point: any, index: number) => {
+            let isPointInLine = false;
+            if (index === annotationCoordinates.length - 1) {
+              // Specific case for the last point that does not have a next point
+              isPointInLine = pointDistance(point, selectionPosition) < width;
+            } else {
+              isPointInLine =
+                geojs.util.distance2dToLineSquared(
+                  selectionPosition,
+                  point,
+                  annotationCoordinates[index + 1]
+                ) < width;
+            }
+            return isIn || isPointInLine;
+          },
+          false
+        );
+      } else {
+        return geojs.util.pointInPolygon(
+          selectionPosition,
+          annotationCoordinates
+        );
+      }
+    } else {
+      // If the selection annotation type is "Polygon"
+      // Check if the tested annotation (independently from its type)
+      // is in the defined polygon
+      return annotation.coordinates.reduce((isIn: boolean, point: any) => {
+        const isPointIn = geojs.util.pointInPolygon(
+          point,
+          selectionAnnotationCoordinates
+        );
+        return isIn || isPointIn;
+      }, false);
+    }
+  }
+
+  private getSelectedAnnotationsFromTool(selectAnnotation: any) {
+    const selectLocation = {
+      XY: this.xy,
+      Z: this.z,
+      Time: this.time
+    };
+
+    const coordinates = selectAnnotation.coordinates();
+    const type = selectAnnotation.type();
+
+    // Get general information from the map.
+    // When working with pointAnnotation, unitsPerPixels is necessary to
+    // compute the right value of the radius.
+    const unitsPerPixel = this.getMapUnitsPerPixel();
+
+    // Get selected annotations.
+    // Only select annotations that are located at the same XY and Z
+    return this.annotations
+      .filter((annotation: IAnnotation) => {
+        return (
+          (selectLocation.XY === annotation.location.XY ||
+            this.store.unrollXY) &&
+          (selectLocation.Z === annotation.location.Z || this.store.unrollZ) &&
+          (selectLocation.Time === annotation.location.Time ||
+            this.store.unrollT)
+        );
+      })
+      .filter((annotation: IAnnotation) => {
```

Comment:
why split in 2 `filter()` calls instead of one ?
```
this.annotations.filter((annotation: IAnnotation) => {
 return sameLocation(selectLocation, annotation, this.store.unrollXY, this.store.unrollZ, this.store.unrollT) &&
            this.shouldSelectAnnotation(type, coordinates, annotation, unitsPerPixel);
 }
```